packages/squiggle-lang/__tests__/SquiggleLibrary/SquiggleLibrary_FunctionRegistryLibrary_test.res

```diff
@@ -87,6 +87,8 @@ describe("FunctionRegistry Library", () => {
       "Error(Error: Too few samples when constructing sample set)",
     )
 
+    testEvalToBe("Dict.set({a: 1, b: 2}, 'c', 3)", "Ok({a: 1,b: 2,c: 3})")
+    testEvalToBe("d={a: 1, b: 2}; _=Dict.set(d, 'c', 3); d", "Ok({a: 1,b: 2})")
```

Comment:
A comment on what this test is about would be nice. (I understand why it's here and it's a good idea, but it's not immediately obvious.)

Reply:
Added one saying "Immutable" in #1326.